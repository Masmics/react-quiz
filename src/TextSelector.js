import React from 'react';
import PropTypes from 'prop-types';

function TextSelector({ footerText, handleChange }) {
  return (
    <input 
      name="footerText"
      type="text"
      value={footerText}
      onChange={handleChange}
    />
  );
}

TextSelector.propTypes = {
  footerText: PropTypes.string,
  handleChange: PropTypes.func
};

export default TextSelector;

// Container component
// contains state
// Uses the other components
