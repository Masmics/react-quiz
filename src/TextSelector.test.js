import React from 'react';
import { shallow } from 'enzyme';
import TextSelector from './TextSelector';

describe('text selector component', () => {
  it('handles text input', () => {
    const wrapper = shallow(<TextSelector />);
    expect(wrapper).toMatchSnapshot();
  });
});
