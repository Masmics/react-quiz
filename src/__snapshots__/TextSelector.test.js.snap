// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`text selector component handles text input 1`] = `
<input
  name="footerText"
  type="text"
/>
`;
